src/main/java/uk/ac/ebi/eva/evaseqcol/model/IngestionResultEntity.java

```diff
@@ -0,0 +1,38 @@
+package uk.ac.ebi.eva.evaseqcol.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This Entity will hold the information that should be returned
+ * upon ingestion of seqcol objects (given the assembly accession)*/
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class IngestionResultEntity {
+
+    @JsonProperty("assembly_accession")
+    private String assemblyAccession;
+    @JsonProperty("num_inserted_seqcols")
+    private Integer numberOfInsertedSeqcols = 0;
+    @JsonProperty("inserted_seqcols")
+    private List<InsertedSeqColEntity> insertedSeqcols = new ArrayList<>();
+    @JsonProperty("error_message")
+    private String errorMessage = null;
+
+
+    public void addInsertedSeqCol(InsertedSeqColEntity insertedSeqCol) {
+        this.insertedSeqcols.add(insertedSeqCol);
+    }
+
+    /**
+     * Increment the numberOfInsertedSeqcols by one*/
+    public void incrementNumberOfInsertedSeqCols() {
+        this.numberOfInsertedSeqcols += 1;
```

Comment:
On the other hand, if we want to implement something like what Tim suggests, where the list length isn't the same as the count, you should ignore this completely!

Reply:
On the other other hand, maybe we actually want something like [MongoDB's WriteResult](https://www.mongodb.com/docs/manual/reference/method/BulkWriteResult/), in which case we'd list exactly which seqcols were inserted and which were already found, with the accompanying counts... this might be overkill though...

Reply:
Thank you for the feedback @apriltuesday , your suggestions make complete sense.

For now, we made a decision on keeping the current template of the ingestion response entity as it is, but return it **only** when at least one seqcol object has been inserted, otherwise return only status code and error message for other scenarios, to avoid using the PUT method for other purposes than the one it has been designed for, as you highlighted, and that makes sense.

And we'll be having future discussions on the improvements of the returned JSON object, considering this approach,

> On the other other hand, maybe we actually want something like [MongoDB's WriteResult](https://www.mongodb.com/docs/manual/reference/method/BulkWriteResult/), in which case we'd list exactly which seqcols were inserted and which were already found, with the accompanying counts... this might be overkill though...

which has a lot of similarities of what we want to implement.